ui/src/containers/pipelines/EditPipelinePayload.js

```diff
@@ -0,0 +1,194 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import Breadcrumb from "../../components/layout/Breadcrumb";
+import { PayloadEditor } from "../../components/payload_editor/PayloadEditor";
+import {
+  fetchPipeline,
+  updatePipeline,
+  inspectPipeline,
+} from "../../actions/pipelines";
+import { inspectStream } from "../../actions/streams";
+import { fetchFilters } from "../../actions/filters";
+import { fetchTransforms } from "../../actions/transforms";
+import "../../scss/grid.scss";
+import "../../scss/grid/statistics.scss";
+
+class EditPipelinePayload extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      // storing, whether a new column has been added, in the state
+      // is a bit nasty but will speed up the componentDidMount function
+      unpersistedChanges: false,
+      errorMessage: "",
+      editorPipeline: "",
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.getPipelineId = this.getPipelineId.bind(this);
+    this.submitEditorContent = this.submitEditorContent.bind(this);
+  }
+
+  getPipelineId() {
+    return this.props.match.params.id;
+  }
+
+  componentDidMount() {
+    this.props.fetchPipeline(this.getPipelineId()).then(() => {
+      const pipeline = this.props.pipelines.pipeline;
+
+      // Load sample records from the source stream
+      if (pipeline.metadata["stream-in"] !== undefined) {
+        this.props.inspectStream(pipeline.metadata["stream-in"]);
+      }
+
+      this.setState({
+        pipeline: pipeline,
+        editorPipeline: JSON.stringify(pipeline, null, 2),
+      });
+    });
+
+    this.props.fetchFilters();
+    this.props.fetchTransforms();
+  }
+
+  handleChange(editorString) {
+    this.setState({
+      editorPipeline: editorString,
+      unpersistedChanges: true,
+    });
+  }
+
+  submitEditorContent(event) {
+    event.preventDefault();
+    let parsedPipeline = undefined;
+    try {
+      parsedPipeline = JSON.parse(this.state.editorPipeline, null, 2);
+    } catch (syntaxError) {
+      this.setState({
+        errorMessage: syntaxError.message,
+      });
+    }
+
+    if (this.state.unpersistedChanges && parsedPipeline !== undefined) {
+      return this.props
+        .updatePipeline(this.getPipelineId(), parsedPipeline)
```

Comment:
I would suggest redirecting the user to the "ShowPipeline" page after successfully submitting the new pipeline revision, similar to the UX of the other resources.